lib/saluki-event/src/metric/metadata.rs

```diff
@@ -96,10 +96,10 @@ impl OriginEntity {
 /// Metadata includes all information that is not specifically related to the context or value of the metric itself,
 /// such as sample rate and timestamp.
 #[must_use]
-#[derive(Clone, Debug, Default)]
+#[derive(Clone, Debug)]
 pub struct MetricMetadata {
-    sample_rate: Option<f64>,
-    timestamp: Option<u64>,
+    sample_rate: f64,
+    timestamp: u64,
```

Comment:
We've specifically made those both non-optional because:

- it shrinks them from 16 bytes to 8 bytes
- we can just set the default sample rate to 1.0, which is what we would do anyways when unwrapping it and finding it to be `None`
- timestamp being 0 can serve as the `None` value, since we know it's impossible for a real timestamp to actually be 0